### PR TITLE
Functions to return project components, resolutions and issue types

### PR DESCRIPTION
**Pull request by N-M**
Functions to return project components, resolutions and issue types

### CODE REVIEW

File: src/Jira/Api.php [hunk 1]
```diff
@@ -198,6 +198,17 @@ public function editIssue($issue_key, array $params)
...
+	 * @return array
+	 * @since  2.0.0
+	 */
+	public function getAttachmentsMetaInformation()
+	{
+		return $this->api(self::REQUEST_GET, '/rest/api/2/attachment/meta');
+	}
+
+	/**
 	 * Gets attachment.
```
Comment (jpastoor) [inline src/Jira/Api.php]: Isn't getAttachmentsMetaInformation already merged #101? Please remove from this diff

File: src/Jira/Api.php [hunk 2]
```diff
@@ -835,4 +846,42 @@ public function closeIssue($issue_key)
...
+	 * Get all issue types with valid status values for a project.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array
+	 * @since  2.0.0
+	 */
+	public function getAllIssueTypes($project_key) 
+	{
+		return $this->api(self::REQUEST_GET, "/rest/api/2/project/{$project_key}/statuses", array(), true);
```
Comment (jpastoor) [inline src/Jira/Api.php]: Please use sprintf here as well

File: src/Jira/Api.php [hunk 3]
```diff
@@ -835,4 +846,42 @@ public function closeIssue($issue_key)
...
+
+	 /**
+	 * Get all issue types with valid status values for a project.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array
+	 * @since  2.0.0
+	 */
+	public function getAllIssueTypes($project_key)
```
Comment (jpastoor) [inline src/Jira/Api.php]: We already have a getIssueTypes who returns the global list. This one is specific for projects. Maybe rename to getProjectIssueTypes?

Comment (N-M) [inline src/Jira/Api.php]: I agree on everything above it will be corrected and committed asap.

**Comment by jpastoor** [comment]
Cheers, and thank you for contributing

File: src/Jira/Api.php [hunk 4]
```diff
@@ -835,4 +846,42 @@ public function closeIssue($issue_key)
 		return $result;
 	}
 
+	 /**
+	 * Returns project components.
+	 *
+	 * @param string $project_key Project key.
+	 *
+	 * @return array|false
```
Comment (aik099) [inline src/Jira/Api.php]: In which cases the `false` gets returned?

File: src/Jira/Api.php [hunk 5]
```diff
@@ -835,4 +846,42 @@ public function closeIssue($issue_key)
...
+	 * @since  2.0.0
+	 */
+	public function getResolutions() 
+	{
+		return $this->api(self::REQUEST_GET, "/rest/api/2/resolution", array(), true);
+	}
```
Comment (aik099) [inline src/Jira/Api.php]: There must be an empty line after last function in a class.

Please configure CS check locally to see the issue.

File: src/Jira/Api.php (hunk 2 above)
Comment (aik099) [inline src/Jira/Api.php]: And single quotes.

**Comment by aik099** [comment]
@N-M , please create new branch for each PR you send. Don't send PR from `master` branch.

**Comment by N-M** [comment]
I am closing this pull request and resending from new branch.

**Pull request closed without merging**